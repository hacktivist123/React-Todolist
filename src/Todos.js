import React from 'react';

const Todos = ({ todos, deleteTodo }) => {

    const todoList = todos.length !== 0 ? (
        todos.map( (todo, i) => {
            return(
                <div className= "collection-item" key={i}>
                    <span onClick={ () => {deleteTodo(todo.id)}}>{todo.content}</span>
                    
                </div>
            )
        })
    ) : (
        <p className="center">You have No todo's left, Yay!</p>
    )
    return (
        <div className="todos collection">
        { todoList }

        </div>
    )
}

export default Todos;