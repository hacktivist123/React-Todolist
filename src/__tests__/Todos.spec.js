import React from 'react'
import { shallow } from 'enzyme'
import Todos from '../Todos';

describe('Todos', () => {
  it('should render todos correctly with an array of todos', () => {
    const wrapper = shallow(<Todos todos={[{
      id: 1,
      content: 'test'
    }]} deleteTodo={jest.fn()} />);

    wrapper.find('span').simulate('click', () => {});
    expect(wrapper).toMatchSnapshot();
  })
  it('should render todos list with an empty array', () => {
    const wrapper = shallow(<Todos todos={[]} deleteTodo={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  })
});