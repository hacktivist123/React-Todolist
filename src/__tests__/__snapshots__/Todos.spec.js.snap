// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos should render todos correctly with an array of todos 1`] = `
<div
  className="todos collection"
>
  <div
    className="collection-item"
    key="1"
  >
    <span
      onClick={[Function]}
    >
      test
    </span>
  </div>
</div>
`;

exports[`Todos should render todos list with an empty array 1`] = `
<div
  className="todos collection"
>
  <p
    className="center"
  >
    You have No todo's left, Yay!
  </p>
</div>
`;
